velox/functions/lib/Re2Functions.h

```diff
@@ -30,33 +30,118 @@ namespace facebook::velox::functions {
 enum class PatternKind {
   /// Pattern containing wildcard character '_' only, such as _, __, ____.
   kExactlyN,
-  /// Pattern containing wildcard characters ('_' or '%') only with atleast one
+  /// Pattern containing wildcard characters ('_' or '%') only with at least one
   /// '%', such as ___%, _%__.
   kAtLeastN,
   /// Pattern with no wildcard characters, such as 'presto', 'foo'.
   kFixed,
+  /// Pattern with single wildcard chars(_) & normal chars, such as
+  /// '_pr_es_to_'.
+  kRelaxedFixed,
   /// Fixed pattern followed by one or more '%', such as 'hello%', 'foo%%%%'.
   kPrefix,
+  /// kRelaxedFixed pattern followed by one or more '%', such as '_pr_es_to_%',
+  /// '_pr_es_to_%%%%'.
+  kRelaxedPrefix,
   /// Fixed pattern preceded by one or more '%', such as '%foo', '%%%hello'.
   kSuffix,
+  /// kRelaxedFixed preceded by one or more '%', such as '%_pr_es_to_',
+  /// '%%%_pr_es_to_'.
+  kRelaxedSuffix,
   /// Patterns matching '%{c0}%', such as '%foo%%', '%%%hello%'.
   kSubstring,
   /// Patterns which do not fit any of the above types, such as 'hello_world',
   /// '_presto%'.
   kGeneric,
 };
 
-struct PatternMetadata {
-  PatternKind patternKind;
-  // Contains the length of the unescaped fixed pattern for patterns of kind
-  // kFixed, kPrefix, kSuffix and kSubstring. Contains the count of wildcard
-  // character '_' for patterns of kind kExactlyN and kAtLeastN. Contains 0
-  // otherwise.
+// Kind of sub-pattern.
+enum SubPatternKind {
```

Comment:
@xumingming Would you double check that this enum and other structs used only in the implementation are placed inside of 'detail' namespace?

Reply:
@mbasmanova  The reference chain is: `PatternMetadata -> SubPatternMetadata -> SubPatternKind`, PatternMetadata currently is in the header file, because we have test in `Re2FunctionsTest.cpp` to test `determinePatternKind`, so all of them need to be in the header file?

Reply:
Got it. Thank you for explaining.